access/handler.go

```diff
@@ -580,6 +581,194 @@ func (h *Handler) GetAccountAtBlockHeight(
 	}, nil
 }
 
+// GetAccountBalanceAtLatestBlock returns an account balance by address at the latest sealed block.
+func (h *Handler) GetAccountBalanceAtLatestBlock(
+	ctx context.Context,
+	req *access.GetAccountBalanceAtLatestBlockRequest,
+) (*access.AccountBalanceResponse, error) {
+	metadata, err := h.buildMetadataResponse()
+	if err != nil {
+		return nil, err
+	}
+
+	address, err := convert.Address(req.GetAddress(), h.chain)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid address: %v", err)
+	}
+
+	accountBalance, err := h.api.GetAccountBalanceAtLatestBlock(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+
+	return &access.AccountBalanceResponse{
+		Balance:  accountBalance,
+		Metadata: metadata,
+	}, nil
+}
+
+// GetAccountBalanceAtBlockHeight returns an account balance by address at the given block height.
+func (h *Handler) GetAccountBalanceAtBlockHeight(
+	ctx context.Context,
+	req *access.GetAccountBalanceAtBlockHeightRequest,
+) (*access.AccountBalanceResponse, error) {
+	metadata, err := h.buildMetadataResponse()
+	if err != nil {
+		return nil, err
+	}
+
+	address, err := convert.Address(req.GetAddress(), h.chain)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid address: %v", err)
+	}
+
+	accountBalance, err := h.api.GetAccountBalanceAtBlockHeight(ctx, address, req.GetBlockHeight())
+	if err != nil {
+		return nil, err
+	}
+
+	return &access.AccountBalanceResponse{
+		Balance:  accountBalance,
+		Metadata: metadata,
+	}, nil
+}
+
+// GetAccountKeyAtLatestBlock returns an account public key by address and key index at the latest sealed block.
+func (h *Handler) GetAccountKeyAtLatestBlock(
+	ctx context.Context,
+	req *access.GetAccountKeyAtLatestBlockRequest,
+) (*access.AccountKeyResponse, error) {
+	metadata, err := h.buildMetadataResponse()
+	if err != nil {
+		return nil, err
+	}
+
+	address, err := convert.Address(req.GetAddress(), h.chain)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid address: %v", err)
+	}
+
+	keyByIndex, err := h.api.GetAccountKeyAtLatestBlock(ctx, address, uint64(req.GetIndex()))
+	if err != nil {
+		return nil, err
+	}
+
+	accountKey, err := convert.AccountKeyToMessage(*keyByIndex)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid key: %v", err)
```

Comment:
this would be an internal error since the data came from the node's DB and should be correct
```suggestion
		return nil, status.Errorf(codes.Internal, "failed to encode account key: %v", err)
```